src/software/geom/algorithms/step_along_perimeter_test.cpp

```diff
@@ -0,0 +1,181 @@
+#include "software/geom/algorithms/step_along_perimeter.h"
+
+#include <gtest/gtest.h>
+
+#include "software/geom/point.h"
+#include "software/geom/polygon.h"
+
+
+// Test case: Traveling a distance of zero should return the starting point
+TEST(StepAlongPerimeterTest, ZeroDistanceReturnsStartPoint)
+{
+    Polygon polygon({{0, 0}, {0, 1}, {1, 1}, {1, 0}});
```

Comment:
great tests! but can you make the lower_snake_case variable convention change here as well

Reply:
Following up on this. Test case names (e.g. `ZeroDistanceReturnsStartPoint`) should be snake_case